backend/FwHeadless/Controllers/MediaFileController.cs

```diff
@@ -0,0 +1,150 @@
+using FwHeadless.Services;
+using LexCore.Entities;
+using LexData;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.Extensions.Options;
+
+namespace FwHeadless.Controllers;
+
+[ApiController]
+[Route("/api/media")]
+// TODO: RequireScope? Which scope should we require here? Decide on auth story
+public class MediaFileController : ControllerBase
+{
+    [HttpGet("/{fileId}")]
+    async Task<Results<Ok<FileStream>, NotFound>> GetFile(
+        Guid fileId,
+        ProjectLookupService projectLookupService,
+        IOptions<FwHeadlessConfig> config,
+        LexBoxDbContext lexBoxDb)
+    {
+        var mediaFile = await lexBoxDb.Files.FindAsync(fileId);
+        if (mediaFile is null) return TypedResults.NotFound();
+        var projectId = mediaFile.ProjectId;
+        var project = await lexBoxDb.Projects.FindAsync(projectId);
+        if (project is null) return TypedResults.NotFound();
+        var projectFolder = config.Value.GetProjectFolder(project.Code, projectId);
+        var filePath = Path.Join(projectFolder, mediaFile.Filename);
+        var file = System.IO.File.OpenRead(filePath);
+        return TypedResults.Ok(file);
+    }
+
+    [HttpPut("/{fileId}")]
+    async Task<Results<Ok, BadRequest, NotFound>> PutFile(
+        Guid fileId,
+        Stream body,
+        ProjectLookupService projectLookupService,
+        IOptions<FwHeadlessConfig> config,
+        LexBoxDbContext lexBoxDb)
+    {
+        var mediaFile = await lexBoxDb.Files.FindAsync(fileId);
+        if (mediaFile is null) return TypedResults.NotFound();
+        var projectId = mediaFile.ProjectId;
+        var project = await lexBoxDb.Projects.FindAsync(projectId);
+        if (project is null) return TypedResults.NotFound();
+        var projectFolder = config.Value.GetProjectFolder(project.Code, projectId);
+        var filePath = Path.Join(projectFolder, mediaFile.Filename);
+        int size = 0;
+        try
+        {
+            size = await WriteFileToDisk(filePath, body);
+        }
+        catch (ArgumentOutOfRangeException)
+        {
+            return TypedResults.BadRequest();
+        }
+        if (mediaFile.Metadata is null)
+        {
+            mediaFile.Metadata = new FileMetadata()
+            {
+                SizeInBytes = size,
+                Filename = mediaFile.Filename
+            };
+        }
+        else
+        {
+            mediaFile.Metadata.SizeInBytes = size;
+        }
+        mediaFile.UpdateUpdatedDate();
+        lexBoxDb.SaveChanges();
+        return TypedResults.Ok();
+        // TODO: Extract out common code between this and Post, e.g. for file-size handling, because right now POST has some features that PUT does not
+    }
+
+    [HttpPost]
+    async Task<Results<Created<PostFileResult>, NotFound, BadRequest>> PostFile(
+        [FromForm] Guid fileId,
+        [FromForm] Guid projectId,
+        [FromForm] string filename,
+        [FromForm] Stream file,
```

Comment:
does this work? I think you need to use IFormFile. There's a lot that's not relevant but there's some useful stuff here https://learn.microsoft.com/en-us/aspnet/core/mvc/models/file-uploads?view=aspnetcore-9.0#file-upload-scenarios